feat: add s3 fdw (#67)
supabase/wrappers: docs/s3.md

> [AWS S3](https://aws.amazon.com/s3/) is an object storage service offering industry-leading scalability, data availability, security, and performance. The S3 wrapper only supports data read at this moment and below are the 2 data file formats supported:

Reviewer: ```suggestion
[AWS S3](https://aws.amazon.com/s3/) is an object storage service offering industry-leading scalability, data availability, security, and performance. The S3 wrapper is under development. It is read-only and supports 2 file formats:
```

```sql
create extension if not exists wrappers;

create foreign data wrapper s3_wrapper
  handler s3_fdw_handler
  validator s3_fdw_validator;
```

### Server 

Next, we need to create a server for the FDW to hold options and credentials.

#### Auth (Supabase)

If you are using the Supabase platform, this is the recommended approach for securing your AWS credentials.

For example, to store AWS access key id to `vault_access_key_id` in Vault:

Create a secure key using pgsodium
```sql
select pgsodium.create_key(name := 'vault_access_key_id');
```

Save your credential in Vault and retrieve the `key_id`
```sql
insert into
  vault.secrets (secret, key_id)
values 
  (
    'AKIAXXXXXX',
    (select id from pgsodium.valid_key where name = 'vault_access_key_id')
  );
returning
  key_id;
```

You can follow the same steps to store `vault_secret_access_key` in Vault.

Then create the foreign server

```sql
create server s3_server
  foreign data wrapper s3_wrapper
  options (
    vault_access_key_id '<your vault_access_key_id from above>',
    vault_secret_access_key '<your vault_secret_access_key_id from above>',
    aws_region 'us-east-1'
  );
```

> #### Auth (Insecure)

Reviewer: ```suggestion
#### Auth (Non-Supabase)
```
